src/device-registry/utils/create-device.js

```diff
@@ -98,57 +98,54 @@ const createDevice = {
       });
     }
   },
-  generateQR: async (request) => {
+  generateQR: async (request, callback) => {
     try {
-      let { include_site } = request.query;
-      let responseFromListDevice = await createDevice.list(request);
-      if (responseFromListDevice.success) {
-        let deviceBody = responseFromListDevice.data;
+      const { include_site } = request.query;
+      const responseFromListDevice = await createDevice.list(request);
+      logObject("responseFromListDevice", responseFromListDevice);
+      if (responseFromListDevice.success === true) {
+        const deviceBody = responseFromListDevice.data;
         if (isEmpty(deviceBody)) {
-          return {
+          return callback({
```

Comment:
just using callback function @BenjaminSsempala

Reply:
Thanks @BenjaminSsempala , I was bug hunting and made some changes when things were not working out. But later realised that the changes I made did not fix the issue but was just another way to implement things.

In summary, this is just an alternative way of implementing the same thing. Perhaps neater but no strong motivation for the refactor. It just happened "by mistake" as I was troubleshooting problems which never existed. :)